### PR TITLE
Added single block data from Blockchain.com

### PR DESCRIPTION
**Pull request by noubre**
Added the API for pulling single block JSON data from Blockchain.com. The motivation for this is expand the available API's for BTC blockchain data on the terminal, several more are in the process of being tested. No new dependencies are needed for this addition.

Screenshot 2022-10-26 000048

# How has this been tested?

- [ ] Make sure affected commands still run in terminal
- [ ] Ensure the SDK still works
- [ ] Check any related reports

# Checklist:

- [ ] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

### CODE REVIEW

File: openbb_terminal/cryptocurrency/onchain/blockchain_model.py [hunk 1]
```diff
@@ -27,7 +27,44 @@ def _make_request(endpoint: str) -> dict:
 
     url = f"https://api.blockchain.info/{endpoint}"
     response = requests.get(
-        url, headers={"Accept": "application/json", "User-Agent": "GST"}
+        url,
+        headers={
+            "Accept": "application/json",
+            "User-Agent": "GST",
```
Comment (jose-donato) [inline openbb_terminal/cryptocurrency/onchain/blockchain_model.py:33]: GST -> OBB

File: openbb_terminal/cryptocurrency/onchain/blockchain_model.py [hunk 2]
```diff
@@ -27,7 +27,44 @@ def _make_request(endpoint: str) -> dict:
...
+    dict:
+        dictionary with response data
+    """
+
+    url = f"https://blockchain.info/{endpoint}"
+    response = requests.get(
+        url,
+        headers={
+            "Accept": "application/json",
+            "User-Agent": "GST",
```
Comment (jose-donato) [inline openbb_terminal/cryptocurrency/onchain/blockchain_model.py:65]: OBB

File: openbb_terminal/cryptocurrency/onchain/blockchain_view.py [hunk 3]
```diff
@@ -129,3 +132,57 @@ def display_btc_confirmed_transactions(
...
+        Hash of the block you are looking for.
+    export : str
+        Export dataframe data to csv,json,xlsx file
+    """
+
+    df = blockchain_model.get_btc_single_block(blockhash)
+    df.rename(index={0: "Values"}, inplace=True)
+    df_data = df.copy()
+
+    df_essentials = df[[
```
Comment (jose-donato) [inline openbb_terminal/cryptocurrency/onchain/blockchain_view.py:155]: @DidierRLopes should this transformation be done here or in the model?

File: website/content/terminal/crypto/onchain/btcblockdata/_index.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+Displays json data for a single block. [Source: https://api.blockchain.info/]
+
+```
+arguments:
+  --blockhash BLOCKHASH         Hash of the block you are targeting  
+  --export {csv,json,xlsx}      Export dataframe data to csv,json,xlsx file (default: )
+  -h, --help                    Show this help message (default: False)
+```
+
+![](../../../../../../../../../Dropbox/Screenshot 2022-10-26 000048.jpg)
```
Comment (jose-donato) [inline website/content/terminal/crypto/onchain/btcblockdata/_index.md:15]: the image is only locally on your computer, can you use the image that's in the pr? - https://user-images.githubusercontent.com/57280701/201509706-a7c94df8-e15e-4b99-97bd-4cefa52edb02.jpg

**Review by jose-donato — CHANGES_REQUESTED** [review]
Still need to test it locally but looking nice!! request a few minor changes

Thank you for this 🤝 🚀

**Comment by noubre** [comment]
Thanks! I'll get on those changes this weekend.

File: openbb_terminal/cryptocurrency/onchain/blockchain_model.py (hunk 1 above)
Comment (noubre) [reply openbb_terminal/cryptocurrency/onchain/blockchain_model.py]: Done

File: openbb_terminal/cryptocurrency/onchain/blockchain_model.py (hunk 2 above)
Comment (noubre) [reply openbb_terminal/cryptocurrency/onchain/blockchain_model.py]: Done

File: openbb_terminal/cryptocurrency/onchain/blockchain_view.py (hunk 3 above)
Comment (noubre) [reply openbb_terminal/cryptocurrency/onchain/blockchain_view.py:155]: Done

File: website/content/terminal/crypto/onchain/btcblockdata/_index.md (hunk 4 above)
Comment (noubre) [reply website/content/terminal/crypto/onchain/btcblockdata/_index.md]: Done

**Comment by DidierRLopes** [comment]
Hey @noubre,

Sorry for the delay to review on our side. As per our discussion, we are currently doing a feature freeze for a very big release that will occur next week! Once that is out of the way, we will test this PR and give feedback.

Thanks for contributing! 🙏🏽

**Comment by noubre** [comment]
@DidierRLopes No problem, thanks!

**Comment by jose-donato** [comment]
Hey, going through this again!

Thanks again, this is amazing and will be interesting for the future refactor im planning for the crypto menu.

Requests:
1. add autocompletion for the arguments
2. work if user passes the hash directly without --blockhash e.g., `btcblockdata 0000000000000000000385b3a5aa6ba0c4e9d2bf80a54bda9251b96d5f50b2a3`
3. improve error message
image

4. add command description to file `openbb_terminal/miscellaneous/i18n/en.yml`
5. add command source to file `openbb_terminal/miscellaneous/data_sources_default.json`
6. add column name to 1st column header of the table - name should be `Metric` and second column name should be `Value` and not `Values`

TODO in the future:
- Add a feature to get latest blocks (why? i had to go to the web to check a block hash for example)


Let me know if you need any help with any of this.

**Comment by noubre** [comment]
@jose-donato I'll right on those changes. Thanks! By the way, the new terminal 2.0 looks great!

**Comment by jose-donato** [comment]
@noubre thanks for the quickness 😄 
let me know here or on discord in what you need help

**Comment by jose-donato** [comment]
`openbb_terminal\miscellaneous\library\trail_map.csv` we just need to add the func to this trailmap to make it available from the sdk. apart from that (and the linting/tests) everything look good :)

**Comment by noubre** [comment]
> `openbb_terminal\miscellaneous\library\trail_map.csv` we just need to add the func to this trailmap to make it available from the sdk. apart from that (and the linting/tests) everything look good :)

**Review by jose-donato — APPROVED** [review]
looks good to me! 👍

**Comment by noubre** [comment]
> This was a rough one due to the launch of terminal 2.0 
> 
> Thanks for sticking around and for your help @noubre 🙌 

Thanks! Can't wait to do it again. Thanks for your help!

**Pull request merged**